app/src/organisms/Devices/HistoricalProtocolRunDrawer.tsx

```diff
@@ -0,0 +1,311 @@
+  )
+  const { data } = useAllCsvFilesQuery(run.protocolId ?? '')
+  const allProtocolDataFiles = data != null ? data.data.files : []
+  const uniqueLabwareOffsets = allLabwareOffsets?.filter(
+    (offset, index, array) => {
+      return (
+        array.findIndex(
+          firstOffset =>
+            firstOffset.location.slotName === offset.location.slotName &&
+            firstOffset.definitionUri === offset.definitionUri
+        ) === index && !isEqual(offset.vector, { x: 0, y: 0, z: 0 })
+      )
+    }
+  )
+
+  const deckCalibrationData = useDeckCalibrationData(robotName)
+    .deckCalibrationData
+  const lastModifiedDeckCal =
+    deckCalibrationData != null && 'lastModified' in deckCalibrationData
+      ? deckCalibrationData.lastModified
+      : null
+  const protocolDetails = useMostRecentCompletedAnalysis(run.id)
+  const enableCsvFile = useFeatureFlag('enableCsvFile')
+
+  const isOutOfDate =
+    typeof lastModifiedDeckCal === 'string' &&
+    uniqueLabwareOffsets != null &&
+    uniqueLabwareOffsets.length > 0 &&
+    new Date(lastModifiedDeckCal).getTime() >
+      new Date(
+        uniqueLabwareOffsets[uniqueLabwareOffsets?.length - 1].createdAt
+      ).getTime()
+  const outOfDateBanner = isOutOfDate ? (
+    <Banner
+      type="warning"
+      marginTop={SPACING.spacing8}
+      iconMarginLeft={SPACING.spacing4}
+    >
+      <Flex flexDirection={DIRECTION_COLUMN}>
+        <StyledText as="p" fontWeight={TYPOGRAPHY.fontWeightSemiBold}>
+          {t('data_out_of_date')}
+        </StyledText>
+        <StyledText as="p">{t('robot_was_recalibrated')}</StyledText>
+      </Flex>
+    </Banner>
+  ) : null
+
+  const protocolFilesData =
+    allProtocolDataFiles.length === 0 ? (
+      <InfoScreen contentType="noFiles" t={t} backgroundColor={COLORS.grey35} />
+    ) : (
+      <Flex flexDirection={DIRECTION_COLUMN} gridGap={SPACING.spacing4}>
+        <StyledText>{t('protocol_files')}</StyledText>
+        <Flex
+          direction={DIRECTION_COLUMN}
+          justifyContent={JUSTIFY_FLEX_START}
+          alignItems={ALIGN_END}
+          gridGap={SPACING.spacing24}
+          color={COLORS.grey60}
+          padding={`${SPACING.spacing4} ${SPACING.spacing12}`}
+        >
+          <Box width="33%">
+            <StyledText
+              as="p"
+              datatest-id="RecentProtocolRun_Drawer_fileNameTitle"
+            >
+              {t('name')}
+            </StyledText>
+          </Box>
+          <Box width="33%">
+            <StyledText
+              as="p"
+              datatest-id="RecentProtocolRun_Drawer_fileDateTitle"
+            >
+              {t('date')}
+            </StyledText>
+          </Box>
+          <Box width="34%">
+            <StyledText
+              as="p"
+              datatest-id="RecentProtocolRun_Drawer_fileDownloadTitle"
+            >
+              {t('download')}
+            </StyledText>
```

Comment:
![Screenshot 2024-06-21 at 10 48 23 AM](https://github.com/Opentrons/opentrons/assets/474225/cf41499f-19ed-491e-aa58-c0c8500f83a9)

for vertical alignmen

```suggestion
<Flex gridGap={SPACING.spacing4} alignItems={ALIGN_CENTER}>
                      <StyledText as="p">{t('download')}</StyledText>
                      <Icon name="download" size="1.25rem" />
                    </Flex>
```

Reply:
<img width="434" alt="Screenshot 2024-06-21 at 10 54 08 AM" src="https://github.com/Opentrons/opentrons/assets/474225/59d2b1cd-6d55-4c49-bc62-b533412ebe0a">